code/frameworks/react-vite/package.json

```diff
@@ -51,7 +51,7 @@
     "@rollup/pluginutils": "^5.0.2",
     "@storybook/builder-vite": "workspace:*",
     "@storybook/react": "workspace:*",
-    "@vitejs/plugin-react": "^3.0.1",
+    "@vitejs/plugin-react": "^3.0.1 || ^4.2.0",
```

Comment:
This is a bit more complex, unfortunately.
@vitejs/plugin-react@4 requires Vite 4.2, but we still support Vite 3. If we define it like this, the latest version is always installed. See the discussion in https://github.com/storybookjs/storybook/pull/24901

Reply:
Please tell me what to do, right now this is my only option to getting the CI green.

Reply:
Maybe @IanVS has some advice?

Reply:
I'm trying to bundle it in now, that way the mismatching peerDep thing won't be a surface.

Reply:
Can we add a yarn resolution/override for CI?  That is what we have to tell our users to do to avoid the peer dependency warning as well, until we can drop support for vite < 4.2 (in SB 8 perhaps?).

The other option we could consider is adopting a bit of a "semver-lite" approach, whereby we allow some types of small breaking changes during minor releases, like dropping support for old versions of vite when using the fallback plugins.

I'm not sure that bundling will work, because it needs to be installed in a way that vite can find it, right?